src/libcore/iter.rs

```diff
@@ -1698,7 +1698,7 @@ impl<B, I: DoubleEndedIterator, F> DoubleEndedIterator for Map<I, F> where
 {
     #[inline]
     fn next_back(&mut self) -> Option<B> {
-        self.iter.next_back().map(|a| (self.f)(a))
+        self.iter.next_back().map(self.f)
```

Comment:
How does this work? Closures are not `Copy`, and you only have mutable access, can't move out of `self` - did you mean to write `&mut self.f`?